core/primitives/src/state_proof.rs

```diff
@@ -0,0 +1,37 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_primitives_core::hash::CryptoHash;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, BorshSerialize, BorshDeserialize, PartialEq, Eq)]
```

Comment:
Just want to make sure that we need both serde serialization and borsh.  For this and for the structs below.

Reply:
I don't know how much functionality we want to export from primitives. I gave the highest possible. But it can be restricted to serialization only if we only consider what the RPC node needs.

Reply:
I looked into it.  This proof is part of QueryResponse which is part of RoutedMessageBody which is borsh-serialised since it is signed (AFAIU).  On the other hand, On the other hand, QueryRequest can also be made over JSON via ‘call’ query command.

Reply:
The more derives, the more code we have to generate and the slower it is to build the project.  My opinion would be to derive the minimal set.  And additional derives can be added on a need basis.

Reply:
I think we need them all

```
#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
pub struct StateItem{}

#[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
pub struct ViewStateResult {}
```

Reply:
Got it.  Thanks for looking it up.